go-controller/pkg/ovn/egressgw.go

```diff
@@ -191,7 +191,7 @@ func (oc *Controller) addPodExternalGWForNamespace(namespace string, pod *kapi.P
 		return fmt.Errorf("failed to ensure namespace locked: %v", err)
 	}
 	tmpPodGWs := oc.getRoutingPodGWs(nsInfo)
-	tmpPodGWs[pod.Name] = egress
+	tmpPodGWs[makePodGWKey(pod)] = egress
```

Comment:
hmm is this the actual fix? this only fixes the log line right? I thought we said we need to add a check to see if the same pod comes up again or not in https://github.com/trozet/ovn-kubernetes/blob/10bf7136db147dcd7be8b4d983555b95af42c9b2/go-controller/pkg/ovn/namespace.go#L79 ?
Example:

1. foo_podA got added to tmpPodGWs cache
2. the actual transact following this failed
3. we retry the add, and we try to add foo_podA again? Already exists in cache and hence we error out no?

Reply:
nvm, the bug happened because we had two entries for the same pod in tmp cache.